### PR TITLE
[luci/import] CircleConstNodeBuilder::build

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise CircleConstNodeBuilder::build to load extended Buffer.

### CODE REVIEW

**Comment by seanshpark** [comment]
- for #13678
- from #13615

File: compiler/luci/import/src/Nodes/CircleConst.cpp [hunk 1]
```diff
@@ -160,7 +164,45 @@ CircleNode *CircleConstNodeBuilder::build(TensorIndex tensor_index,
   const auto c_buffer = const_tensor->buffer();
   const auto r_buffer = r_buffers[c_buffer];
   assert(r_buffer != nullptr);
-  const auto buffer = wrap(r_buffer->data());
+  // temporary buffer to provide raw data from file
+  // must have life time same or longer than 'buffer' variable
+  std::vector<uint8_t> temp_buffer;
+  // const auto buffer = wrap(r_buffer->data());
```
Comment (jinevening) [inline compiler/luci/import/src/Nodes/CircleConst.cpp:170]:
```suggestion
```

File: compiler/luci/import/src/Nodes/CircleConst.cpp [hunk 2]
```diff
@@ -160,7 +164,45 @@ CircleNode *CircleConstNodeBuilder::build(TensorIndex tensor_index,
...
+  luci::VectorWrapper<uint8_t> buffer(nullptr);
+  if (r_buffer->offset() > 1)
+  {
+    uint32_t r_size = static_cast<uint32_t>(r_buffer->size());
```
Comment (jinevening) [inline compiler/luci/import/src/Nodes/CircleConst.cpp:174]: How about throwing an exception if `r_buffer` > max of uint32_t?

File: compiler/luci/import/src/Nodes/CircleConst.cpp [hunk 3]
```diff
@@ -160,7 +164,45 @@ CircleNode *CircleConstNodeBuilder::build(TensorIndex tensor_index,
   const auto c_buffer = const_tensor->buffer();
   const auto r_buffer = r_buffers[c_buffer];
   assert(r_buffer != nullptr);
-  const auto buffer = wrap(r_buffer->data());
+  // temporary buffer to provide raw data from file
+  // must have life time same or longer than 'buffer' variable
+  std::vector<uint8_t> temp_buffer;
+  // const auto buffer = wrap(r_buffer->data());
+  luci::VectorWrapper<uint8_t> buffer(nullptr);
+  if (r_buffer->offset() > 1)
```
Comment (jinevening) [inline compiler/luci/import/src/Nodes/CircleConst.cpp:172]: Q) Is it ok to use `r_buffer->data()` for offset == 1?

Comment (seanshpark) [reply compiler/luci/import/src/Nodes/CircleConst.cpp:171]: no. `1` for size and offset is invalid.
I'll add some checks here :)

Comment (jinevening) [reply compiler/luci/import/src/Nodes/CircleConst.cpp:171]: I thought that condition might be `if (r_buffer->offset() > 0)`.

Comment (seanshpark) [reply compiler/luci/import/src/Nodes/CircleConst.cpp:171]: `offset`, `size` with `1` is made for initial values for the extended Buffer.
these value must be updated to real offset and size after the full stream of file data is generated, that is we'll do something like `seek(offset)` and `read(size)` later in import.

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**